pkg/apiserver/resthandler.go

```diff
@@ -676,6 +676,13 @@ func UpdateResource(r rest.Updater, scope RequestScope, typer runtime.ObjectType
 			}
 		}
 
+		canDelete := func() bool {
```

Comment:
I'm not sure I agree that we'll evolve any more code than we already have in rest handler that is already "incorrect".  Admission control is currently implemented on internal typed objects, so it's no worse.  Patch is a special case of the more general problem - it's doing retries itself instead of relying on the platform to do it.

If you'd prefer, we can just drop this entire check in this PR (the security aspect), and Jordan or myself will come in in the next few days and do the refactor to properly give admission the data it needs to restrict.  That then sets the stage for a correct / non-hacky check to be added, as well as prepping for field level authorization.